ccmlib/cluster.py

```diff
@@ -121,6 +123,91 @@ def set_install_dir(self, install_dir=None, version=None, verbose=False):
 
         return self
 
+    def actively_watch_logs_for_error(self, on_error_call, interval=1):
+        """
+        Begins a thread that repeatedly scans system.log for new errors, every interval seconds.
+        (The first pass covers the entire log contents written at that point,
+        subsequent scans cover newly appended log messages).
+        Reports new errors, by calling the provided callback with an OrderedDictionary
+        mapping node name to a list of error lines.
+        Returns the thread itself, which should be .join()'ed to wrap up execution,
+        otherwise will run until the main thread exits.
+        """
+        class LogWatchingThread():
+            """
+            This class is embedded here for now, because it is used only from
+            within Cluster, and depends on cluster.nodelist().
+            """
+
+            def __init__(self, cluster):
+                self.executor = ThreadPoolExecutor(max_workers=1)
+                self.thread = None
+                self.cluster = cluster
+                self.req_stop_event = threading.Event()
+                self.done_event = threading.Event()
+                self.log_positions = defaultdict(int)
+
+            def start(self):
+                self.thread = self.executor.submit(self.run)
+
+            def scan(self):
+                errordata = OrderedDict()
```

Comment:
There is no need to use `OrderedDict` because starting with `Python3.6`, the key insertion order is saved.

```
result = {key:key for key in range(10)}
list(range(10)) == list(result.keys())
True
```

Reply:
in theory CCM support python versions lower then 3.6 (in practice, we probably broken it long time ago)